src/main/kotlin/blackjack/domain/Game.kt

```diff
@@ -0,0 +1,56 @@
+package blackjack.domain
+
+class Game(
+    private val players: List<Player>,
+    private val dealer: Dealer,
+    private val observer: GameObserver
+) {
+
+    private val participants = mutableListOf<Participant>().apply {
+        add(dealer)
+        addAll(players)
+    }
+
+    init {
+        require(players.isNotEmpty()) { "플레이어는 최소 한 명 이상이어야 합니다." }
+    }
+
+    fun startGame() {
+        participants.forEach { initialDeal(it) }
+        observer.onGameStarted(participants)
+    }
+
+    private fun initialDeal(participant: Participant) = repeat(INITIAL_CARDS) {
+        participant.receiveCard(dealer.dealCard())
+    }
+
+    fun playerTurn(shouldHit: (Player) -> Boolean) {
+        players.forEach { player ->
+            while (shouldHit(player)) {
+                player.receiveCard(dealer.dealCard())
+                if (player.cards.calculateScore() <= PLAYER_MAXIMUM_SCORE) {
```

Comment:
여기서 디미터 법칙을 위반하고 있네요.

cards의 확장 함수로 calculateScore를 호출하고 있는 형태인데요, 현재 카드 목록 출력을 위해 cards의 노출이 필요한 상황이라 큰 의미는 없겠지만 가급적이면 cards를 숨기고 아래와 같은 형태로 구현하는게 바람직하다고 보여집니다.

```suggestion
                if (player.calculateScore() <= PLAYER_MAXIMUM_SCORE) {
```